GitHubDeployments.ps1

```diff
@@ -0,0 +1,567 @@
+    }
+
+    $params = @{
+        'UriFragment' = $uriFragment
+        'Method' = 'Get'
+        'Description' = "Getting $EnvironmentName from $RepositoryName"
+        'AccessToken' = $AccessToken
+        'TelemetryEventName' = $MyInvocation.MyCommand.Name
+        'TelemetryProperties' = $telemetryProperties
+    }
+
+    Write-Debug -Message ('UriFragment: ' + $params.UriFragment)
+
+    $result =  Invoke-GHRestMethod @params
+
+    if ($PSBoundParameters.ContainsKey('EnvironmentName'))
+    {
+        return ($result | Add-GitHubDeploymentEnvironmentAdditionalProperties)
+    }
+    else
+    {
+        return ($result.environments | Add-GitHubDeploymentEnvironmentAdditionalProperties)
+    }
+}
+
+filter Add-GitHubDeploymentEnvironmentAdditionalProperties
+{
+<#
+    .SYNOPSIS
+        Adds type name and additional properties to ease pipelining to GitHub Deployment Environment
+        objects.
+
+    .PARAMETER InputObject
+        The GitHub object to add additional properties to.
+
+    .PARAMETER TypeName
+        The type that should be assigned to the object.
+
+    .PARAMETER OwnerName
+        Owner of the repository.  This information might be obtainable from InputObject, so this
+        is optional based on what InputObject contains.
+
+    .PARAMETER RepositoryName
+        Name of the repository.  This information might be obtainable from InputObject, so this
+        is optional based on what InputObject contains.
+
+    .INPUTS
+        [PSCustomObject]
+
+    .OUTPUTS
+        GitHub.Repository
+#>
+    [CmdletBinding()]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseSingularNouns", "", Justification="Internal helper that is definitely adding more than one property.")]
+    param(
+        [Parameter(
+            Mandatory,
+            ValueFromPipeline)]
+        [AllowNull()]
+        [AllowEmptyCollection()]
+        [PSCustomObject[]] $InputObject,
+
+        [ValidateNotNullOrEmpty()]
+        [string] $TypeName = $script:GitHubDeploymentEnvironmentTypeName,
+
+        [string] $OwnerName,
+
+        [string] $RepositoryName
+    )
+
+    foreach ($item in $InputObject)
+    {
+        $item.PSObject.TypeNames.Insert(0, $TypeName)
+
+        if (-not (Get-GitHubConfiguration -Name DisablePipelineSupport))
+        {
+            $repositoryUrl = [String]::Empty
+            if ([String]::IsNullOrEmpty($item.html_url))
+            {
+                if ($PSBoundParameters.ContainsKey('OwnerName') -and
+                    $PSBoundParameters.ContainsKey('RepositoryName'))
+                {
+                    $repositoryUrl = (Join-GitHubUri -OwnerName $OwnerName -RepositoryName $RepositoryName)
+                }
+            }
+            else
+            {
+                $elements = Split-GitHubUri -Uri $item.html_url
+                $repositoryUrl = Join-GitHubUri @elements
+            }
+
+            if (-not [String]::IsNullOrEmpty($repositoryUrl))
```

Comment:
Seems like it would make sense to also process the `DeploymentBranchPolicy` value so that it can be interpreted the way the user originally configured it (similar to what you did for Team permissions back in #300).

We should also further process the Team and User objects returned within the `permissions` object by sending them through `Add-*AdditionalProperties` as well.

Reply:
I've added `ReviewerUser`, `ReviewerTeam`, `WaitTimer` and `DeploymentBranchPolicy` to the `'GitHub.DeploymentEnvironment'` output object and updated the unit tests to check for these.